src/hotspot/share/runtime/globals.hpp

```diff
@@ -551,6 +551,11 @@ const int ObjectAlignmentInBytes = 8;
   product(bool, HeapDumpAfterFullGC, false, MANAGEABLE,                     \
           "Dump heap to file after any major stop-the-world GC")            \
                                                                             \
+  product(uint, FullGCHeapDumpLimit, 0, MANAGEABLE,                         \
+          "Limit the number of heap dumps triggered by "                    \
+          "HeapDumpBeforeFullGC or HeapDumpAfterFullGC "                    \
+          "(0 means no limit)" )                                            \
```

Comment:
nit: should be no space before the last paren.

Reply:
I was wondering if it is worth having HeapDumpBeforeFullGC and HeapDumpAfterFullGC also mention FullGCHeapDumpLimit. Just something simple like "Also see FullGCHeapDumpLimit." Not necessary, but if you think it would be useful then please add.

Reply:
> nit: should be no space before the last paren.

fixed.

Reply:
> I was wondering if it is worth having HeapDumpBeforeFullGC and HeapDumpAfterFullGC also mention FullGCHeapDumpLimit. Just something simple like "Also see FullGCHeapDumpLimit." Not necessary, but if you think it would be useful then please add.

Make sense. Added.